jaxkern/approximations/rff.py

```diff
@@ -0,0 +1,35 @@
+from ..base import AbstractKernel
+from ..computations import BasisFunctionComputation
+from jax.random import KeyArray
+from typing import Dict, Any
+
+
+class RFF(AbstractKernel):
+    def __init__(self, base_kernel: AbstractKernel, num_basis_fns: int) -> None:
+        self._check_valid_base_kernel(base_kernel)
+        self.base_kernel = base_kernel
+        self.num_basis_fns = num_basis_fns
+        # Set the computation engine to be basis function computation engine
+        self.compute_engine = BasisFunctionComputation
+        # Inform the compute engine of the number of basis functions
+        self.compute_engine.num_basis_fns = num_basis_fns
+
+    def init_params(self, key: KeyArray) -> Dict:
+        base_params = self.base_kernel.init_params(key)
+        n_dims = base_params["lengthscale"].shape[0]
```

Comment:
If we ever do spectral decomp for a kernel without "lengthscale" then we are in trouble.

Reply:
How likely is that? Personally, I’m OK with it for now. The workaround would be adding an attribute to each kernel that tells us whether or not it is parameterised by a lengthscale. If no, then a vector of ones would be created in the RFF approximation. To me, this feels a little messy.